eos-tech-support/eos-write-metrics-collector-usb

```diff
@@ -0,0 +1,107 @@
+#!/bin/bash -e
+
+# Detect the plugged in USB storage and prompt users
+# that the contents of the storage will be erased
+# and reformat the whole disk to a single FAT32
+# partition.
+# Then copy both eos-metrics-collector and the
+# eos-metrics-uploader with .exe extension for
+# the users to execute to collect offline metrics
+# data.
+
+USB_DISK_NUM=0
+
+prompt_input()
+{
+    echo "
+	 "
+    read -p "Please enter the disk number you want to create for metrics? [] "
+    response="${REPLY,,}" # to lower
+    if [[ ! "$response" =~ ^[1-9]$ ]]; then
+        echo " Please re-input your selection "
+        prompt_input
+    fi
+    if [ $((response)) -gt $((USB_DISK_NUM)) ]; then
+        echo " Please re-input your selection "
+        prompt_input
+    fi
+
+    USER_CHOICE=$response
+}
+
+for device in /sys/block/*
+do
+    if udevadm info --query=property --path=$device | grep -q ^ID_BUS=usb
+    then
+        DISK[$USB_DISK_NUM]="/dev/${device##*/}"
+        ((USB_DISK_NUM=USB_DISK_NUM+1))
+        echo "$USB_DISK_NUM) / ${DISK[$((USB_DISK_NUM-1))]}"
+        echo "`lsblk -o name,label /dev/${device##*/}`"
+    fi
+done
+
+if [ $USB_DISK_NUM -eq 0 ]; then
+    echo "Please plug in the USB storage device."
+    exit 1
+fi
+
+prompt_input
+
+TARGET_DISK=${DISK[$((USER_CHOICE-1))]}
+
+echo "disk name $TARGET_DISK"
+MOUNTPOINT=$(grep "$TARGET_DISK" /proc/mounts | cut -f2 -d' ' | tr '\n' ' ')
```

Comment:
This check does not work if the existing filesystem label has spaces on it. But you don't need to look into `/proc/mounts` at all, you can unmount with udisksctl, as you do on line 106 here.

Reply:
Also, you need to make sure you are unmounting all partitions of the target disk.

Reply:
Actually this is OK for unmouting multiple partitions. I'll put the `/dev/sda1 /dev/sda2...` etc in one line by`tr '\n' ' '` and `umount` can unmount multiple partitions. But there's problem if the one of the device is busy. So I replace the `umount` with the `udiskctrl`.  But I still use `/proc/mounts` for parsing because the output of `lsblk` and `udiskctl` is not that easy to parse. So I keep it as-as.